src/SprykerEco/Yves/Braintree/Handler/BraintreeHandler.php

```diff
@@ -61,7 +61,7 @@ public function __construct(CurrencyPluginInterface $currencyPlugin, BraintreeCo
      */
     public function addPaymentToQuote(Request $request, QuoteTransfer $quoteTransfer)
     {
-        $paymentSelection = $quoteTransfer->getPayment()->getPaymentSelection();
+        $paymentSelection = $quoteTransfer->getPayment()->getPaymentSelectionOrFail();
```

Comment:
Fixed `getPayment() -> getPaymentOrFail()`. 

Kept `OrFail` for `getPaymentSelection()`. Otherwise, phpstan complains about type incompatibility. Looks like it’s not a BC-break, since with an empty payment selection the code below would fail anyway.